### DISCUSSION TITLE
Require Output File Variable Description Documentation

### DISCUSSION THREAD

**Discussion by Lise-Li** [NASA-LIS/LISF#1438] [Q&A] [answered]
Hi, LISF Community!

I am running Public testcase walkthrough, each time I followed the steps and generated the result file, I want to see the exact meaning of the variables in it, where can I find the documentation of the description of the variables in the file generated in each step? If such documentation is not available, what tools are available to help understand the meaning of each variable in the generated file? For example, I generated the file cdf_noah36.nc in step4, this file contains 4 variables, where can I find the exact meaning of these four variables?

Looking forward to your reply!
Thank you!
Lise Li

**Comment by dmocko** [comment]
Hi @Lise-Li 

The four variables in this file are:
```
SoilMoist_xrange=>xrange  1  y,z,t,x  SoilMoist_xrange
SoilMoist_mu=>mu          1    z,t,x  SoilMoist_mu
SoilMoist_sigma=>sigma    1    z,t,x  SoilMoist_sigma
SoilMoist_CDF=>cdf        1  y,z,t,x  SoilMoist_CDF
```

The codes that calculates these variables are found in (respectively):
https://github.com/NASA-LIS/LISF/blob/master/ldt/core/LDT_DrangeMod.F90
https://github.com/NASA-LIS/LISF/blob/master/ldt/core/LDT_MuMod.F90
https://github.com/NASA-LIS/LISF/blob/master/ldt/core/LDT_SigmaMod.F90
https://github.com/NASA-LIS/LISF/blob/master/ldt/core/LDT_CDFMod.F90

I don't know any more than that, as I not a data assimilation expert.

@LIS-navari , @yyoon4 , @sujayvkumar - Can one of you please help answer this question?

**Reply by LIS-navari** [reply, marked as answer]
Hi @Lise-Li and @dmocko,

The LISF user guide should suffice for new users. However, those keen on understanding all intermediate processes and variables should refer to the subroutines for comprehensive insights into variable meanings. In your case, examining the output file (cdf_noah36.nc) will provide essential information within the NetCDF file, aiding in understanding each variable.

Here are key details from the file:

dimensions:
    ngrid = 52597;
    ntimes = 12;
    nbins = 100;
    SoilMoist_levels = 1;

variables:
    float SoilMoist_xrange(nbins, SoilMoist_levels, ntimes, ngrid);
    float SoilMoist_mu(SoilMoist_levels, ntimes, ngrid);
    float SoilMoist_sigma(SoilMoist_levels, ntimes, ngrid);
    float SoilMoist_CDF(nbins, SoilMoist_levels, ntimes, ngrid);

ngrid = 52597: Indicates the number of grid cells in your domain, where 52597 represents the grid cells in the NLDAS domain.
ntimes = 12: Represents 12 months, signifying a soil moisture Cumulative Distribution Function (CDF) for each month. If ntimes = 1, it implies a lump CDF representing all soil moisture events throughout the year. Please refer to ldt.config (Temporal resolution of CDFs: "monthly" # yearly | monthly).
nbins = 100: Represents the number of bins in the CDF. Please refer to ldt.config (Number of bins to use in the CDF: 100).
SoilMoist_levels = 1: Represents the number of levels in our analysis. Here, the CDF computation is for the surface layer.
Hope this clarifies the variables' significance!

SoilMoist_xrange: Represents the x range in each bin.
SoilMoist_CDF: Represents the CDF value for each bin. Utilizing these variables, you can plot the soil moisture CDF for each grid cell for each month.
Moreover:

SoilMoist_mu and SoilMoist_sigma: Provide the mean and standard deviation of soil moisture for each grid cell for each month.

Best,
Mahdi
